test/gnosisSafePersonalEdition.js

```diff
@@ -49,10 +53,61 @@ contract('GnosisSafePersonalEdition', function(accounts) {
         let executorDiff = await web3.eth.getBalance(executor) - executorBalance
         console.log("    Executor earned " + web3.fromWei(executorDiff, 'ether') + " ETH")
         assert.ok(executorDiff > 0)
-    })
+    });
+
+    it('should deposit and withdraw 1 ETH paying with token', async () => {
+        let token = await safeUtils.deployToken(accounts[0]);
+        let executorBalance = (await token.balances(executor)).toNumber();
+        await token.transfer(gnosisSafe.address, 10000000, {from: accounts[0]});
+        await web3.eth.sendTransaction({from: accounts[0], to: gnosisSafe.address, value: web3.toWei(1.1, 'ether')})
+        await safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, {
+          gasToken: token.address
+        });
+
+        await safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, {
+          gasToken: token.address
+        })
+        let executorDiff = (await token.balances(executor)).toNumber() - executorBalance;
+        console.log("    Executor earned " + executorDiff + " Tokens")
+        assert.ok(executorDiff > 0);
+    });
+
+    it('should fail when depositing 1 ETH paying with token due to token transfer fail', async () => {
+        let mockContract = await MockContract.new();
+        let mockToken = MockToken.at(mockContract.address);
+        let transferData = await mockToken.contract.transfer.getData(executor, 91861);
+        await mockContract.givenRevertAny(transferData.slice(0,10));
+        await web3.eth.sendTransaction({from: accounts[0], to: gnosisSafe.address, value: web3.toWei(1.1, 'ether')})
```

Comment:
can we set balance to 0.5 and check at the end of the test that it is still 0.5

Reply:
I have an assert that checks if the balance is 0.5 in the new commit